articles/user-guide/host-programs.md

```diff
@@ -0,0 +1,563 @@
+Additionally, it's no problem if your callable includes [documentation comments](xref:microsoft.quantum.guide.filestructure#documentation-comments) before its declaration, the `@EntryPoint()` attribute can be simply placed above them.
+
+### Callable arguments
+
+So far, we've only considered an operation that takes no inputs.
+Suppose we wanted to perform a similar operation, but on multiple qubits---the number of which is provided as an argument.
+Such an operation could be written as
+```qsharp
+    operation MeasureSuperpositionArray(n : Int) : Result[] {
+        using (qubits = Qubit[n]) {              // allocate a register of n qubits
+            ApplyToEach(H, qubits);              // apply H to each qubit in the register
+            return ForEach(MResetZ, qubits);     // perform MResetZ on each qubit, returns the resulting array
+        }
+    }
+```
+where the returned value is an array of the measurement results.
+Note that [`ApplyToEach`](xref:microsoft.quantum.canon.applytoeach) and [`ForEach`](xref:microsoft.quantum.arrays.foreach) are in the [`Microsoft.Quantum.Canon`](xref:microsoft.quantum.canon) and [`Microsoft.Quantum.Arrays`](xref:microsoft.quantum.arrays) namespaces, requiring additional `open` statements for each.
+
+If we move the `@EntryPoint()` attribute to precede this new operation (note there can only be one such line in a file), attempting to run it with simply `dotnet run` results in an error message which indicates what additional command line options are required, and how to express them.
+
+The general format for the command line is actually `dotnet run [options]`, and callable arguments are provided there.
+In this case, the argument `n` is missing, and it shows that we need to provide the option `-n <n>`. 
+To run `MeasureSuperpositionArray` for `n=4` qubits, we therefore use
+
+```bash
+dotnet run -n 4
+```
+
+yielding an output similar to
+
+```bash
+[Zero,One,One,One]
+```
+
+This of course extends to multiple arguments.
+
+> [!NOTE]
+> Argument names defined in `camelCase` are slightly altered by the compiler to be accepted as Q# inputs. 
+> For example, if instead of `n`, we used the name `numQubits` above, then this input would be provided in the command line via `--num-qubits 4` instead of `-n 4`.
+
+The error message also provides other options which can be used, including how to change the target machine.
+
+### Different target machines
+
+As the outputs from our operations thus far have been the expected results of their action on real qubits, it's clear that the default target machine from the command line is the full-state quauntum simulator, `QuantumSimulator`.
+However, we can instruct callables to be run on a specific target machine with the option `--simulator` (or the shorthand `-s`).
+
+For example, we could run it on [`ResourcesEstimator`](xref:microsoft.quantum.machines.resources-estimator):
+
+```bash
+dotnet run -n 4 -s ResourcesEstimator
+```
+
+The printed output is then
+
+```bash
```

Comment:
Would it make sense as a future improvement to have this in a tab next to Jupyter Notebook's visual output?

Reply:
@cgranade referring to the output code blocks on this page specifically? Personally I think it would run a risk of accidentally conflating the two methods, especially as the audience here will be a lot of new/prospective users. 

I did consider potentially putting each of the methods in its own tab, but ultimately felt that approach was more fitting for the install instructions, etc., where users have already "picked a lane". Supposing a lot of readers on this page will be here to _decide_ which method they'd like to start using, I thought it best to leave things in distinct sections which mirror each other in what's being done. That way each section can stand fully on its own (useful for people who have picked a lane and come looking for more details on that method), but at the same time be an easily comprehensible overview for those who would like to read it through and see how the methods compare.